nodeup/pkg/model/kubelet.go

```diff
@@ -771,16 +768,28 @@ func (b *KubeletBuilder) kubeletNames() ([]string, error) {
 		return append(addrs, name), nil
 	}
 
-	cloud := b.Cloud.(awsup.AWSCloud)
+	addrs := []string{b.InstanceID}
+	sess := session.Must(session.NewSession())
+	metadata := ec2metadata.New(sess)
 
-	result, err := cloud.EC2().DescribeInstances(&ec2.DescribeInstancesInput{
-		InstanceIds: []*string{&b.InstanceID},
-	})
-	if err != nil {
-		return nil, fmt.Errorf("error describing instances: %v", err)
+	if localHostname, err := metadata.GetMetadata("local-hostname"); err == nil {
```

Comment:
We used to have an issue (IIRC) where local-hostname did not match PrivateDNSName, when users had a VPC with a custom DNS suffix.  However, I think that we're now OK because (1) we use the instance ID as the node name now, rather than PrivateDNSName and  (2) kubeletNames seems to _mostly_ be ignored now anyway, as we usually issue the cert via kops-controller.